packages/fuels-signers/src/wallet.rs

```diff
@@ -109,8 +100,14 @@ impl From<WalletError> for Error {
 }
 
 impl Wallet {
-    /// Create a new wallet from a private key.
-    pub fn new_from_private_key(private_key: SecretKey, provider: Provider) -> Self {
+    pub fn new(provider: Option<Provider>) -> Self {
```

Comment:
the naming of the function should specify that a new wallet is created _at random_, otherwise one might think that the wallet is just a Rust instance.

Reply:
If the user explicitly want to use the private key, then sure that functionality can be kept. 264 is more about removing dealing with private keys then the user doesn't actually need them, e.g. if they're setting up a wallet then they only need to sign things, not the private key itself.

Reply:
AFAIK the `new_from_private_key` function is still here right?

Reply:
It's there, but private